vignettes/design-principles.Rmd

```diff
@@ -0,0 +1,42 @@
+---
+title: "Design Principles for linelist"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Design Principles for linelist}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+This vignette outlines the design decisions that have been taken during the development of the `{linest}` R package, and provides some of the reasoning, and possible pros and cons of each decision.
```

Comment:
Remove {} from package name for consistency with rest of document? I think the backticks help distinguish it as a package name so could also use them for other "linelist" occurrences.

Reply:
Fixed in [`ee556b4` (#112)](https://github.com/epiverse-trace/linelist/pull/112/commits/ee556b4a74777236141baf3fbda35388a2f5a6d1).